packages/shared/src/services/batch/ScopeResolver.ts

```
@@ -0,0 +1,64 @@
+        }
+
+        case 'contains': {
+            const searchText = scope.config.searchText
+            if (!searchText) return false
+            const content = message.content ?? ''
+            return content.toLowerCase().includes(searchText.toLowerCase())
+        }
+
+        default:
+            // Exhaustive check — TypeScript will error if new cases are added
+            const _exhaustive: never = scope.type
```

Biome (2.5.0)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

(lint/correctness/noSwitchDeclarations)